test/hts-precompile/utils.js

```diff
@@ -668,6 +671,158 @@ class Utils {
     const txResponse = await signTx.execute(signerClient)
     await txResponse.getReceipt(signerClient)
   }
+
+  /**
+   * @dev Constructs a key conforming to the IHederaTokenService.TokenKey type
+   *
+   * @param keyType ADMIN | KYC | FREEZE | WIPE | SUPPLY | FEE | PAUSE
+   *                See https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L128
+   *                for more information
+   *
+   * @param keyValyeType INHERIT_ACCOUNT_KEY | CONTRACT_ID | ED25519 | SECP256K1 | DELEGETABLE_CONTRACT_ID
+   *
+   * @param key bytes value, public address of an account, or boolean
+   *            See https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L92
+   *                     for more information
+   */
+  static constructIHederaTokenKey(keyType, keyValueType, key) {
+    // sanitize params
+    if (
+      (keyType !== 'ADMIN' &&
+        keyType !== 'KYC' &&
+        keyType !== 'FREEZE' &&
+        keyType !== 'WIPE' &&
+        keyType !== 'SUPPLY' &&
+        keyType !== 'FEE' &&
+        keyType !== 'PAUSE') ||
+      (keyValueType !== 'INHERIT_ACCOUNT_KEY' &&
+        keyValueType !== 'CONTRACT_ID' &&
+        keyValueType !== 'ED25519' &&
+        keyValueType !== 'SECP256K1' &&
+        keyValueType !== 'DELEGETABLE_CONTRACT_ID')
+    ) {
+      return
+    }
+
+    return {
```

Comment:
Might be nice to create a const outside of this method that has the default values for all of this.
Then in a `switch` statement based on `keyValueType` you set the appropriate field.
This would simplify it and mean you can simplify the `if` above

Reply:
Fantastic idea! Updating the improvement soon!